packages/@lwc/ssr-runtime/src/render.ts

```diff
@@ -99,19 +103,31 @@ type GenerateMarkupFnVariants =
     | GenerateMarkupFnAsyncNoGen
     | GenerateMarkupFnSyncNoGen;
 
+interface ComponentWithGenerateMarkup {
+    [SYMBOL__GENERATE_MARKUP]: GenerateMarkupFnVariants;
+}
+
 export async function serverSideRenderComponent(
     tagName: string,
-    compiledGenerateMarkup: GenerateMarkupFnVariants,
-    props: Properties,
+    Component: GenerateMarkupFnVariants | ComponentWithGenerateMarkup,
+    props: Properties = {},
     mode: 'asyncYield' | 'async' | 'sync' = 'asyncYield'
 ): Promise<string> {
+    if (typeof tagName !== 'string') {
+        throw new Error(`tagName must be a string, found: ${tagName}`);
+    }
```

Comment:
Just a handy-dandy friendly error message. Adding this to the PR free of charge.